apps/publish/content/common.py

```diff
@@ -266,6 +268,25 @@ def _validate(self, original, updates):
         if len(validation_errors) > 0:
             raise ValidationError(validation_errors)
 
+    def _raise_if_unpublished_related_items(self, original):
+        archive_service = get_resource_service('archive')
+        publishing_warnings_confirmed = request.args.get('publishing_warnings_confirmed')
+        if (not config.PUBLISH_ASSOCIATED_ITEMS
+                and not publishing_warnings_confirmed
+                and original.get(ASSOCIATIONS)
+                and self.publish_type in [ITEM_PUBLISH, ITEM_CORRECT]):
+            for key, associated_item in original.get(ASSOCIATIONS).items():
+                if associated_item and archive_service._is_related_content(key):
+                    item = archive_service.find_one(req=None, _id=associated_item.get('_id'))
+                    if item.get('state') not in PUBLISH_STATES:
+                        error_msg = json.dumps({
+                            'id': 'publishing_warnings_confirmed',
+                            'message': _('There are unpublished related \
```

Comment:
would rather make it a list of warnings so it's more flexible, eg:
```py
json.dumps({
    'warnings': [_('...')],
})
```
not sure we need that id for anything, but if so it would be good to have a constant for it and use it in both places